### DISCUSSION TITLE
Nested Parser with matching components

### DISCUSSION THREAD

**Discussion by petersktang** [pointfreeco/swift-parsing#319] [Q&A]
```
\begin{subequations}
\begin{equation}
  \operatorname{min}_{a,b,c} 
  \frac{1}{2}\mathbf{w}^{T}\mathbf{w} + C \sum_{i=1}^{l}\xi_{i} 
\end{equation}    
\begin{equation*}
  y_{i}\left(\mathbf{w}^{T}\phi(x_{i})+b\right)
\end{equation*}
\end{subequations}
```
How to develop a parser for the above latex math which has matching `begin` and `end` component? There are quite a few other begin/end blocks and can be nested within one another.

**Comment by petersktang** [comment]
closed

**Comment by stephencelis** [comment]
@petersktang I'm not sure if you still need help here, but the library comes with a couple examples of recursive parsers that are used to parse nested data:

- JSON parser: https://github.com/pointfreeco/swift-parsing/blob/main/Sources/swift-parsing-benchmark/JSON.swift
- Arithmetic parser: https://github.com/pointfreeco/swift-parsing/blob/main/Sources/swift-parsing-benchmark/Arithmetic.swift
